Demo/src/main/java/com/paypal/android/ui/payment/paymentbutton/PaymentButton.kt

```diff
@@ -0,0 +1,462 @@
+package com.paypal.android.ui.payment.paymentbutton
+
+import android.content.Context
+import android.content.res.TypedArray
+import android.graphics.Canvas
+import android.os.Build
+import android.util.AttributeSet
+import android.util.TypedValue
+import android.view.Gravity
+import android.view.LayoutInflater
+import android.view.View
+import android.widget.ImageView
+import android.widget.LinearLayout
+import android.widget.ProgressBar
+import android.widget.TextView
+import androidx.annotation.RequiresApi
+import androidx.core.content.ContextCompat
+import androidx.core.content.res.use
+import com.google.android.material.shape.CutCornerTreatment
+import com.google.android.material.shape.MaterialShapeDrawable
+import com.google.android.material.shape.RoundedCornerTreatment
+import com.google.android.material.shape.ShapeAppearanceModel
+import com.paypal.android.R
+
+@Suppress("TooManyFunctions")
+@RequiresApi(Build.VERSION_CODES.M)
+abstract class PaymentButton<C : PaymentButtonColor> @JvmOverloads constructor(
+    context: Context,
+    attributeSet: AttributeSet? = null,
+    defStyleAttr: Int = 0
+) : LinearLayout(context, attributeSet, defStyleAttr) {
+
+    /**
+     * Signals that the backing shape has changed and may require a full redraw.
+     */
+    private var shapeHasChanged = false
+
+    private var shapeAppearanceModel: ShapeAppearanceModel = ShapeAppearanceModel()
+        set(value) {
+            field = value
+            materialShapeDrawable = materialShapeDrawable.apply {
+                shapeAppearanceModel = field
+            }
+        }
+
+    private var materialShapeDrawable: MaterialShapeDrawable = MaterialShapeDrawable()
+        set(value) {
+            field = value
+            background = field
+        }
+
+    private val colorLuminance: PaymentButtonColorLuminance
+        get() = color.luminance
+
+    /**
+     * Updates the color of the Payment Button with the provided [PaymentButtonColor].
+     */
+    abstract var color: C
+
+    /**
+     * Provides the current [PaymentButtonEligibilityStatus]. This value can change after the button
+     * has been rendered, those changes can be observed on by using either
+     * [onEligibilityStatusChanged] or [paymentButtonEligibilityStatusChanged].
+     */
+    private var eligibilityStatus: PaymentButtonEligibilityStatus = PaymentButtonEligibilityStatus.Loading
```

Comment:
I don't think we'll want to keep any of the eligibility logic in the payment buttons. Would you be able to remove everything eligibility related?

Reply:
you are completely right, changes are done now 👍